src/Features/Lsif/Generator/Generator.cs

```diff
@@ -251,6 +251,45 @@ SymbolKind.RangeVariable or
                     {
                         var definitionResultsId = symbolResultsTracker.GetResultIdForSymbol(declaredSymbol, Methods.TextDocumentDefinitionName, () => new DefinitionResult(idFactory));
                         lsifJsonWriter.Write(new Item(definitionResultsId.As<DefinitionResult, Vertex>(), lazyRangeVertex.Value.GetId(), documentVertex.GetId(), idFactory));
+
+                        // If this declared symbol also implements an interface member, we count this as a definition of the interface member as well.
+                        // Note in C# there are estoeric cases where a method can implement an interface member even though the containing type does not
+                        // implement the interface, for example in this case:
+                        //
+                        //     interface I { void M(); }
+                        //     class Base { public void M() { } }
+                        //     class Derived : Base, I { }
+                        //
+                        // We don't worry about supporting these cases here.
+                        var implementedMembers = declaredSymbol.ExplicitOrImplicitInterfaceImplementations();
+
+                        foreach (var implementedMember in implementedMembers)
+                            MarkImplementationOfSymbol(implementedMember);
+
+                        // If this overrides a method, we'll also mark it the same way. We want to chase to the base virtual method, skipping over intermediate
+                        // methods so that way all overrides of the same method point to the same virtual method
```

Comment:
interesting - I would have thought we would have pointed to intermediate members which themselves point back to the base in a chain.  This loses the intermediate member information, but I have no idea if that is OK or not (not sure how this is used).

Reply:
We discussed this and decided to go with this way for now. We saw two potential downsides of having the full chain:

1. It's not clear what happens if somebody removes their method of an override in the middle of a chain, since then the ends of the chain are disconnected.
2. It means a find references in the cloud has to connect all the chains together, as opposed to doing a faster lookup.

So we're going this way for now.